packages/vue/src/pinia.ts

```diff
@@ -0,0 +1,87 @@
+import { addBreadcrumb, getClient, getCurrentScope, getGlobalScope } from '@sentry/core';
+import { addNonEnumerableProperty } from '@sentry/utils';
+
+// Inline PiniaPlugin type
+type PiniaPlugin = (context: {
+  store: {
+    $id: string;
+    $state: unknown;
+    $onAction: (callback: (context: { name: string; after: (callback: () => void) => void }) => void) => void;
+  };
+}) => void;
+
+type SentryPiniaPluginOptions = {
+  attachPiniaState?: boolean;
+  actionTransformer: (action: any) => any;
+  stateTransformer: (state: any) => any;
+};
+
+export const createSentryPiniaPlugin: (options?: SentryPiniaPluginOptions) => PiniaPlugin = (
+  options: SentryPiniaPluginOptions = {
+    attachPiniaState: true,
+    actionTransformer: action => action,
+    stateTransformer: state => state,
+  },
+) => {
+  const plugin: PiniaPlugin = ({ store }) => {
+    options.attachPiniaState &&
+      getGlobalScope().addEventProcessor((event, hint) => {
+        try {
+          // Get current timestamp in hh:mm:ss
+          const timestamp = new Date().toTimeString().split(' ')[0];
+          const filename = `pinia_state_${store.$id}_${timestamp}.json`;
+
+          hint.attachments = [
+            ...(hint.attachments || []),
+            {
+              filename,
+              data: JSON.stringify(store.$state),
```

Comment:
M: What just came to my mind: the store can contain PII, especially when pinia is used in forms. What do you think of adding the type of the value as the value? Like this:
```
name: string
likes: object
```

Reply:
Since the pinia plugin needs to be added manually I would argue it is fine. Just having the type is basically useless for debugging.

Reply:
Stringifying the state as JSON will only work in some scenarios and fail in many others like Dates, Errors, Maps, Sets, and custom classes. I would recommend using https://github.com/Rich-Harris/devalue like Nuxt does to allow users to provide serializer and revivers. I would even recommend adding a custom one that uses a `toJSON()` method if it exists by default as a convention. I think, but I'm not sure, that Nuxt has something similar.

Reply:
Hi, we are on a tight bundle-size budget so we need to be a bit careful what we pull in as deps.

What we definitely should do is wrap the stringify call in a try-catch because it can throw in certain situations.

Non-serializable data is tricky and I would recommend we do it in a follow-up if at all. Dates get serialized properly with ordinary JSON.stringify, errors, maps, and sets, will become objects (which is fine for now), and stringify already supports toJSON natively.

Reply:
The `toJSON()` was specific to devalue ofc. If users can opt-in to a fully compatible serializable solution, they would be able to put their app in a similar state with the serialized state and debug in dev mode, so I think it's nice to advocate for that! Since Map and Sets are common, it would be a shame to lose them fully by using JSON.stringify()`

Reply:
So, this event processor logic is already in try-catch, so we're safe on that part. We can do a follow-up for serializing. Thanks for the review @posva!